web/frontend/src/components/base.vue

```diff
@@ -78,24 +84,66 @@ const setDirection = (dir: Directions) => {
   direction.value = dir;
 };
 
-const handleBaseActionStop = (name: string) => {
+const stop = () => {
   const req = new baseApi.StopRequest();
-  req.setName(name);
+  req.setName(props.name);
   window.baseService.stop(req, new grpc.Metadata(), displayError);
 };
 
-const baseKeyboardCtl = (name: string, controls: Record<string, boolean>) => {
-  if (Object.values(controls).every((item) => item === false)) {
-    handleBaseActionStop(name);
-    return;
+const digestInput = () => {
+  let linearValue = 0;
+  let angularValue = 0;
+
+  for (const item of pressed) {
+    switch (item) {
+      case Keymap.FORWARD: {
+        linearValue += 1;
+        break;
+      }
+      case Keymap.BACKWARD: {
+        linearValue -= 1;
+        break;
+      }
+      case Keymap.LEFT: {
+        angularValue += 1;
+        break;
+      }
+      case Keymap.RIGHT: {
+        angularValue -= 1;
+        break;
+      }
+    }
   }
 
-  const inputs = computeKeyboardBaseControls(controls);
   const linear = new commonApi.Vector3();
   const angular = new commonApi.Vector3();
-  linear.setY(inputs.linear);
-  angular.setZ(inputs.angular);
-  BaseControlHelper.setPower(name, linear, angular, displayError);
+  linear.setY(linearValue);
+  angular.setZ(angularValue);
+
+  const req = new baseApi.SetPowerRequest();
+  req.setName(props.name);
+  req.setLinear(linear);
+  req.setAngular(angular);
+
+  rcLogConditionally(req);
+  window.baseService.setPower(req, new grpc.Metadata(), displayError);
+};
+
+const handleKeyDown = (key: Keys) => {
+  pressed.add(key);
+  digestInput();
```

Comment:
We re-digest input whenever a keydown happens, or a keyup happens and the length of pressed keys is still greater than zero.